src/ethereum/spurious_dragon/vm/interpreter.py

```diff
@@ -113,12 +115,14 @@ def process_create_message(message: Message, env: Environment) -> Evm:
     # take snapshot of state before processing the message
     begin_transaction(env.state)
 
+    increment_nonce(env.state, message.current_target)
     evm = process_message(message, env)
     if not evm.has_erred:
         contract_code = evm.output
         contract_code_gas = len(contract_code) * GAS_CODE_DEPOSIT
         try:
             evm.gas_left = subtract_gas(evm.gas_left, contract_code_gas)
+            ensure(len(contract_code) <= MAX_CODE_SIZE, OutOfGasError)
```

Comment:
Using `OutOfGasError` here seems a bit misleading. Perhaps we can create a new Exception, let's say, `MaxCodeSizeExceededError` and then change the except clause below to `except(OutOfGasError, MaxCodeSizeExceededError)`.

Reply:
My bad, EIP 170 says that an `OutOfGasError` error needs to be raised.

Reply:
I think there's a trade-off here.

If we introduce another exception type, now we have to be careful in every `except` clause. On the other hand, `OutOfGasError` does seem misleading.

Would making `CodeSizeExceededError` (inline bikeshedding!) a subclass of `OutOfGasError` make more or less sense?

Reply:
I suggest that we do more comprehensive exception reform in a separate PR. There should be a superclasses called things like `FailWithAllGasError`, `InvalidTransactionError`, etc.. from which most other exceptions derive. That way we can avoid long `except:` lists.